questionnaire/routes.test.js

```diff
@@ -628,30 +647,86 @@ describe('Hitting /apply/:section', () => {
                         getSectionContext: () => 'submission',
                         addPrefix: section => `p-${section}`,
-                        processRequest: requestBody => requestBody
+                        processRequest: requestBody => requestBody,
+                        removeSectionIdPrefix: () => 'applicant-fatal-claim'
                     }))
                 });
             });
             it('Should submit', async () => {
                 const currentAgent = request.agent(app);
                 const initialResponse = await currentAgent.get('/apply/start-or-resume');
                 const initialCsrfToken = getCsrfTokenFromResponse(initialResponse.res.text);
+                const initialAnalyticsId = 'urn:uuid:ce66be9d-5880-4559-9a93-df15928be396';
                 const response = await currentAgent.post('/apply/applicant-fatal-claim').send({
-                    _csrf: initialCsrfToken
+                    _csrf: initialCsrfToken,
+                    _analyticsId: initialAnalyticsId
                 });
                 expect(response.res.text).toContain('Application submitted');
             });
             it('Should redirect to you have submitted your application page on refresh after confirmation page', async () => {
                 const currentAgent = request.agent(app);
                 const initialResponse = await currentAgent.get('/apply/start-or-resume');
                 const initialCsrfToken = getCsrfTokenFromResponse(initialResponse.res.text);
+                const initialAnalyticsId = 'urn:uuid:ce66be9d-5880-4559-9a93-df15928be396';
+
                 const response = await currentAgent.post('/apply/applicant-fatal-claim').send({
-                    _csrf: initialCsrfToken
+                    _csrf: initialCsrfToken,
+                    _analyticsId: initialAnalyticsId
                 });
                 expect(response.res.text).toContain('Application submitted');
             });
         });
+        describe('Prevent cross-template edits', () => {
+            beforeEach(() => {
+                setUpCommonMocks({
+                    './form-helper.js': jest.fn(() => ({
+                        getSectionContext: () => 'post',
+                        addPrefix: section => `p-${section}`,
+                        processRequest: requestBody => requestBody,
+                        removeSectionIdPrefix: () => 'applicant-fatal-claim'
+                    })),
+                    '../account/account-service': () => {
+                        return jest.fn(() => ({
+                            getOwnerId: () => '123ownerid',
+                            isAuthenticated: req => !!req?.session?.analyticsId
+                        }));
+                    }
+                });
+            });
+            it("Should redirect authenticated users to the dashboard if the session analyticsId doesn't match the analyticsId on the page", async () => {
```

Comment:
Could you organise these tests in the same manor as the the existing one. e.g.:

````
describe('Prevent cross-template edits', () => {
    describe('Authenticated', () => {
    describe('Hitting /apply/start-or-resume', () => {
            describe('session analytics ID and form field match ', () => {
                 it('Should redirect to dashboard', async () => {
                    ....
                });
            });
            describe('session analytics ID and form field mismatch', () => {
                it('Should redirect to first page if the application', async () => {
                    ....
                });
            });
        });
    });
});
````

These tests could arguably be included in the `'Hitting /apply/start-or-resume'` describe block at the top of the file

Reply:
I've added in the extra describe blocks but kept the tests where they are. The apply/start-or-resume is not actually integral to the test it is just to get a hit to generate the CSRF token as in the other tests